hpx/parallel/segmented_algorithms/transform.hpp

```diff
@@ -0,0 +1,876 @@
+                        parallel::util::detail::handle_remote_exceptions<
+                            ExPolicy
+                        >::call(r, errors);
+                        auto ft = r.back().get();
+                        auto olast = traits1::compose(send, std::get<0>(ft));
+                        auto odest = traits2::compose(sdest, std::get<1>(ft));
+                        return std::make_pair(olast, odest);
+                    },
+                    std::move(segments)));
+        }
+
+        ///////////////////////////////////////////////////////////////////////
+        // segmented implementation
+        template <typename ExPolicy, typename SegIter, typename OutIter, typename F,
+            typename Proj>
+        typename util::detail::algorithm_result<ExPolicy,
+            hpx::util::tagged_pair<tag::in(SegIter), tag::out(OutIter)>
+        >::type
+        transform_(ExPolicy&& policy, SegIter first, SegIter last, OutIter dest,
+            F && f, Proj && proj, std::true_type)
+        {
+            typedef parallel::execution::is_sequenced_execution_policy<
+                    ExPolicy
+                > is_seq;
+            typedef util::detail::algorithm_result<ExPolicy,
+                hpx::util::tagged_pair<tag::in(SegIter), tag::out(OutIter)>
+                > result;
+            if (first == last)
+            {
+                return result::get(std::make_pair(std::move(last),
+                    std::move(dest)));
+            }
+
+            typedef hpx::traits::segmented_iterator_traits<SegIter>
+                iterator_traits1;
+            typedef hpx::traits::segmented_iterator_traits<OutIter>
+                iterator_traits2;
+
+            return hpx::util::make_tagged_pair<tag::in, tag::out>(
+                segmented_transform(transform<std::pair<
+                typename iterator_traits1::local_iterator,
+                typename iterator_traits2::local_iterator> >(),
+                std::forward<ExPolicy>(policy), first, last,
+                dest, std::forward<F>(f), proj, is_seq()));
+        }
+
+        // forward declare the non-segmented version of this algorithm
+        template <typename ExPolicy, typename InIter, typename OutIter, typename F,
+            typename Proj>
+        typename util::detail::algorithm_result<ExPolicy,
+            hpx::util::tagged_pair<tag::in(InIter), tag::out(OutIter)>
+        >::type
+        transform_(ExPolicy&& policy, InIter first, InIter last, OutIter dest,
+            F && f, Proj && proj, std::false_type);
+
+        // Binary transform
+
+        // sequential remote implementation
+        template <typename Algo, typename ExPolicy, typename InIter1,
+            typename InIter2, typename OutIter, typename F,
+            typename Proj1, typename Proj2>
+        static typename util::detail::algorithm_result<ExPolicy,
+            hpx::util::tuple<InIter1,InIter2,OutIter>
```

Comment:
@hkaiser @mcopik  At line 252 of segmented/transform, I convert the tuple to a tagged tuple? Is this enough? I based the implementation on the non segmented transform. So segmented_transform() returns a tuple and transform_() calls make_tagged_tuple on that and returns the result.